js/ui/toolbar.js

```diff
@@ -249,6 +250,19 @@ var Toolbar = ToolbarBase.inherit({
         return itemElement;
     },
 
+    _setIconButtonAppearanceConfig: function(buttons, value) {
+        iteratorUtils.each(buttons, function(index, buttonItem) {
+            Button.getInstance(buttonItem).option("_forceIconButtonAppearance", value);
```

Comment:
I think you should pass the `_forceIconButtonAppearance` option value on `_renderItem`. Just extend `item.options` before `callback`